packages/vite/src/node/plugins/reporter.ts

```diff
@@ -119,32 +119,37 @@ export function buildReporterPlugin(config: ResolvedConfig): Plugin {
       compressedCount = 0
     },
 
-    renderChunk(code, chunk) {
-      for (const id of chunk.moduleIds) {
-        const module = this.getModuleInfo(id)
-        if (!module) continue
-        // When a dynamic importer shares a chunk with the imported module,
-        // warn that the dynamic imported module will not be moved to another chunk (#12850).
-        if (module.importers.length && module.dynamicImporters.length) {
-          // Filter out the intersection of dynamic importers and sibling modules in
-          // the same chunk. The intersecting dynamic importers' dynamic import is not
-          // expected to work. Note we're only detecting the direct ineffective
-          // dynamic import here.
-          const detectedIneffectiveDynamicImport = module.dynamicImporters.some(
-            (id) => !isInNodeModules(id) && chunk.moduleIds.includes(id),
-          )
-          if (detectedIneffectiveDynamicImport) {
-            this.warn(
-              `\n(!) ${
-                module.id
-              } is dynamically imported by ${module.dynamicImporters
-                .map((m) => m)
-                .join(', ')} but also statically imported by ${module.importers
-                .map((m) => m)
-                .join(
-                  ', ',
-                )}, dynamic import will not move module into another chunk.\n`,
-            )
+    renderChunk(code, chunk, options) {
+      if (!options.inlineDynamicImports) {
+        for (const id of chunk.moduleIds) {
+          const module = this.getModuleInfo(id)
+          if (!module) continue
+          // When a dynamic importer shares a chunk with the imported module,
+          // warn that the dynamic imported module will not be moved to another chunk (#12850).
+          if (module.importers.length && module.dynamicImporters.length) {
+            // Filter out the intersection of dynamic importers and sibling modules in
+            // the same chunk. The intersecting dynamic importers' dynamic import is not
+            // expected to work. Note we're only detecting the direct ineffective
+            // dynamic import here.
+            const detectedIneffectiveDynamicImport =
+              module.dynamicImporters.some(
+                (id) => !isInNodeModules(id) && chunk.moduleIds.includes(id),
+              )
+            if (detectedIneffectiveDynamicImport) {
+              this.warn(
+                `\n(!) ${
+                  module.id
+                } is dynamically imported by ${module.dynamicImporters
+                  .map((m) => m)
```

Comment:
`map` seems unnecessary? 🤔

Reply:
Indeed, please create another PR for this if you see a way to simplify it. This PR isn't touching the detection and warning code, only adding a guard.

Reply:
Ok, I'll create a PR to remove it.